bitcoin/src/crypto/sighash.rs

```diff
@@ -1493,14 +1492,14 @@ mod tests {
         let script = addr.script_pubkey();
         let cache = SighashCache::new(&tx);
 
-        let got = cache.legacy_signature_hash(0, &script, 0x01).expect("sighash");
+        let got = cache.legacy_signature_hash(0, &script, EcdsaSighashType::All).expect("sighash");
```

Comment:
This one changed 0x01 to `EcdsaSighashType::All` while the one above changed 0 to `EcdsaSighashType::All`. Clearly one of them is wrong.

Reply:
3031a6e682500f3e5406c24c284c6e6b2fbb3aa9 does not appear to have this issue.

Reply:
It's in b50917af0d9b0b5959baa770746a701a9ee2e373

[here](https://github.com/rust-bitcoin/rust-bitcoin/pull/3217/commits/b50917af0d9b0b5959baa770746a701a9ee2e373#diff-2d1e9ce2e1b21135450e5ddce616f6f431650f6376b37dbb6f85130978fa67d5L1379) the *immutable* variable previously set to 0 is removed and [here](https://github.com/rust-bitcoin/rust-bitcoin/pull/3217/commits/b50917af0d9b0b5959baa770746a701a9ee2e373#diff-2d1e9ce2e1b21135450e5ddce616f6f431650f6376b37dbb6f85130978fa67d5R1383) the variable is replaced with the `All` variant which is equal to 0x01.

At the time I wrote the comment I didn't have time to check if the change in semantics matters for the example. Now I see the `0` was arguably wrong in the first place and it was hidden code so nothing was misleading people anyway. Thus ACK this change. It's the kind that's not-entirely-related that wasn't called out in the commit message.

Reply:
Ooh, good catch. I totally missed that there was semantic content in the hidden code.